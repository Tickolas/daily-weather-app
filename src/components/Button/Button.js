import React from 'react';
import injectSheet from 'react-jss';
import 'chart.js';
import classnames from 'classnames';

const styles = {
    button: {
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        padding: 8,
        fontSize: 16,
        borderRadius: 50,

        '&:focus': {
            outline: 0
        }
    },
    icon: {
        width: 20,
        height: 20,
        marginRight: 8
    }
};

export const Button = ({ classes, children, text, icon, alt, className, ...other }) => (
    <button className={classnames(classes.button, className)} {...other}>
        {icon && <img className={classes.icon} src={icon} alt={alt} />}
        {text}
    </button>
);

export default injectSheet(styles)(Button);
