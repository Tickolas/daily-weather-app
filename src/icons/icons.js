export { default as IconRefresh } from 'icons/refresh.svg';
export { default as IconSnow } from 'icons/snow.svg';
export { default as IconRain } from 'icons/rain.svg';
export { default as IconSun } from 'icons/sun.svg';
export { default as IconCloud } from 'icons/cloud.svg';
export { default as IconUnknownWeather } from 'icons/unknown-weather.svg';
