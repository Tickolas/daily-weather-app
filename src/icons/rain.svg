<svg height="512pt" viewBox="-46 0 512 512" width="512pt" xmlns="http://www.w3.org/2000/svg"><path d="m380 243c0-5.523438-4.476562-10-10-10s-10 4.476562-10 10c0 33.085938-26.914062 60-60 60-5.523438 0-10 4.476562-10 10s4.476562 10 10 10c44.113281 0 80-35.886719 80-80zm0 0"/><path d="m171.664062 340.472656c-5.285156 7.972656-51.664062 78.761719-51.664062 108.527344 0 34.148438 27.476562 63 60 63s60-28.851562 60-63c0-29.765625-46.378906-100.554688-51.664062-108.527344-1.855469-2.792968-4.984376-4.472656-8.335938-4.472656s-6.480469 1.679688-8.335938 4.472656zm48.335938 108.527344c0 22.90625-18.691406 43-40 43s-40-20.09375-40-43c0-15.410156 21.964844-55.628906 40-84.503906 18.035156 28.875 40 69.09375 40 84.503906zm0 0"/><path d="m70 166c-3.285156 0-6.359375 1.613281-8.226562 4.316406-6.324219 9.152344-61.773438 90.445313-61.773438 125.683594 0 38.597656 31.402344 70 70 70s70-31.402344 70-70c0-35.238281-55.449219-116.53125-61.773438-125.683594-1.867187-2.703125-4.941406-4.316406-8.226562-4.316406zm0 180c-27.570312 0-50-22.429688-50-50 0-19.339844 27.964844-68.4375 50-102.070312 22.035156 33.628906 50 82.71875 50 102.070312 0 27.570312-22.429688 50-50 50zm0 0"/><path d="m200 233c0 5.523438-4.476562 10-10 10s-10-4.476562-10-10 4.476562-10 10-10 10 4.476562 10 10zm0 0"/><path d="m308.429688 4.617188c-1.839844-2.875-5.015626-4.617188-8.429688-4.617188s-6.589844 1.742188-8.429688 4.617188c-2.878906 4.511718-70.851562 111.308593-99.226562 186.601562-1.949219 5.167969.664062 10.9375 5.832031 12.882812 5.164063 1.949219 10.933594-.660156 12.882813-5.828124 21.851562-57.992188 69.605468-137.964844 88.96875-169.417969 34.1875 55.285156 99.972656 170.652343 99.972656 214.144531 0 55.140625-44.859375 100-100 100-44.644531 0-84.238281-29.980469-96.289062-72.910156-1.492188-5.316406-7.011719-8.417969-12.332032-6.925782-5.316406 1.492188-8.414062 7.015626-6.925781 12.332032 14.464844 51.519531 61.980469 87.503906 115.546875 87.503906 66.167969 0 120-53.832031 120-120 0-55.121094-83.046875-193.785156-111.570312-238.382812zm0 0"/></svg>