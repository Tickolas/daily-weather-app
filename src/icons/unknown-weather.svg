<?xml version="1.0" encoding="iso-8859-1"?>
<!-- Generator: Adobe Illustrator 19.0.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" x="0px" y="0px"
     viewBox="0 0 206.817 206.817" style="enable-background:new 0 0 206.817 206.817;" xml:space="preserve">
<g>
	<g>
		<g>
			<path d="M165.953,87.498c-1.88-29.879-27.386-53.623-58.468-53.623c-24.661,0-46.371,14.815-54.894,37.152
				c-0.719-0.03-1.44-0.045-2.159-0.045C22.624,70.982,0,93.851,0,121.962c0,28.111,22.624,50.98,50.432,50.98
				c0.538,0,1.073-0.009,1.607-0.026c0.141,0.017,0.285,0.026,0.43,0.026h114.106c0.272,0,0.54-0.033,0.797-0.094
				c22.17-1.485,39.445-20.092,39.445-42.64C206.817,107.122,188.613,88.256,165.953,87.498z M166.372,165.763
				c-0.139,0.007-0.278,0.022-0.414,0.047H53.049c-0.235-0.038-0.474-0.047-0.722-0.042c-0.629,0.028-1.261,0.042-1.895,0.042
				c-23.876,0-43.3-19.671-43.3-43.848s19.424-43.848,43.3-43.848c1.438,0,2.862,0.071,4.266,0.209
				c1.665,0.172,3.223-0.853,3.74-2.441c6.811-20.86,26.521-34.874,49.047-34.874c28.368,0,51.448,22.445,51.448,50.087
				c0,0.978,0.402,1.915,1.114,2.589c0.71,0.674,1.686,1.034,2.645,0.972c0.611-0.033,1.226-0.049,1.846-0.049
				c19.381,0,35.148,15.971,35.148,35.602C199.685,149.172,185.053,164.79,166.372,165.763z"/>
			<path d="M107.114,73.099c-13.84,0-25.1,11.26-25.1,25.1h7.132c0-9.907,8.061-17.968,17.968-17.968
				c9.907,0,17.967,8.061,17.967,17.968c0,9.907-8.06,17.967-17.967,17.967v0.002h-3.705v21.118h7.132v-14.25
				c12.219-1.68,21.672-12.163,21.672-24.836C132.212,84.359,120.952,73.099,107.114,73.099z"/>
			<rect x="103.408" y="140.849" width="7.132" height="7.132"/>
		</g>
	</g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
<g>
</g>
</svg>
