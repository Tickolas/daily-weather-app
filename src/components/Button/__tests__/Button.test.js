import React from 'react';
import { Button } from 'components/Button/Button';
import log from 'loglevel';
import { IconRefresh } from 'icons/icons';

describe('Button', () => {
    describe('Default form', () => {
        it('should match snapshot', () => {
            expect(
                mount(
                    <Button
                        classes={{}}
                        onClick={() => log.info('This is the default button')}
                        text={'Default button'}
                    />
                )
            ).toMatchSnapshot();
        });
        it('should match snapshot with icon', () => {
            expect(
                mount(
                    <Button
                        classes={{}}
                        icon={IconRefresh}
                        onClick={() => log.info('This is a button with an icon')}
                        text={'Icon button'}
                    />
                )
            ).toMatchSnapshot();
        });
    });
});
