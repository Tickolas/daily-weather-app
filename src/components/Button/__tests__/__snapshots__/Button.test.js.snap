// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Default form should match snapshot 1`] = `
<Button
  classes={Object {}}
  onClick={[Function]}
  text="Default button"
>
  <button
    className=""
    onClick={[Function]}
  >
    Default button
  </button>
</Button>
`;

exports[`Button Default form should match snapshot with icon 1`] = `
<Button
  classes={Object {}}
  icon="refresh.svg"
  onClick={[Function]}
  text="Icon button"
>
  <button
    className=""
    onClick={[Function]}
  >
    <img
      src="refresh.svg"
    />
    Icon button
  </button>
</Button>
`;
